### ISSUE TITLE
Delete DNS Api Failed 

### ISSUE THREAD

**Issue by Wolf1098** [infinityofspace/certbot_dns_porkbun#68]
**Describe the bug**
The acme_challenge records do not get removed once validated.

**To Reproduce**
Unknown... I have both wildcard and root domains on 1 line with extra -d options????

**Expected behavior**
The porkbun api doesn't seem to be cleanly handling the delete dns records... unsure why

**Certbot command**
Docker Compose File
```
services:
  porkbun-root:
   image: infinityofspace/certbot_dns_porkbun
   platform: linux/arm64/v8
   hostname: cerbot-porkbun-root
   restart: unless-stopped
   volumes:
     - "/storage/docker/certbot/etc/letsencrypt:/etc/letsencrypt"
     - "/storage/docker/certbot/var/lib/letsencrypt:/var/lib/letsencrypt"
     - "/storage/docker/temp-root:/var/log"
     - "/storage/docker/certbot/porkbun/credentials.json:/tmp/credentials.json"
   command: certonly -v --non-interactive --agree-tos --email sysops@root.xyz --preferred-challenges dns --authenticator dns-porkbun  --dns-porkbun-credentials /tmp/credentials.json --dns-porkbun-propagation-seconds 600 -d root.xyz -d root.ca -d root.dev -d root.download -d "*.root.xyz"  -d "*.root.ca" -d "*.root.dev" -d "*.root.download"

```

**Versions (please complete the following version information):**
 - certbot: `2.4.0`
 - certbot_dns_porkbun: `docker: infinityofspace/certbot_dns_porkbun:latest	Platform linux/arm64/v8 ``(sha256:9a65965e4a72d37cf2c99740291e46a579f4b91f1b31188b93f2e94354883a2c)`

**Error message**
```
2024-03-30 21:59:56,369:DEBUG:certbot._internal.error_handler:Encountered signals: [15]
2024-03-30 21:59:56,370:DEBUG:certbot._internal.error_handler:Calling registered functions
2024-03-30 21:59:56,370:INFO:certbot._internal.auth_handler:Cleaning up challenges
2024-03-30 21:59:56,976:ERROR:certbot._internal.error_handler:Encountered exception during recovery: certbot.errors.PluginError: ERROR: DNS delete api call was not successfully
2024-03-30 21:59:56,977:DEBUG:certbot._internal.error_handler:Calling signal 15
2024-03-30 21:59:56,977:DEBUG:certbot._internal.log:Exiting abnormally:
2024-03-30 21:59:56,981:ERROR:certbot._internal.log:An unexpected error occurred while handling the authorizations.
```

**Additional context**
I have 8 domains using the san feature, 4 of which are the root domains, and 4 are wildcards for the subdomains

**Comment by blotree** [comment]
I've encountered the same error using the plugin on different machines, similarly with multiple domains and wildcards. I've installed this plugin from source in one case, through the Arch User Repository build script in another and none of the components are containerized. 

This isn't a critical error if it happens a couple of times but the renewal process can fail if there are too many TXT records that haven't been cleaned up as no more can be created after a point.

Nor sure where the problem is exactly as making API deletion calls using curl directly or pkb-client directly works as expected.
